### PR TITLE
Add more effects for Philips Hue lights

### PR DESCRIPTION
**Pull request by hanskroner**
Recent firmware updates for Philips Hue lights have added additional lighting effects. This PR updates the library's list to include them.

Additionally, this PR fetches the effects a light reports as supported from the `capabilities` key that was [recently added](https://github.com/dresden-elektronik/deconz-rest-plugin/pull/6316) to deCONZ's REST API. This enables a follow up PR to the Home Assistant integration that would add these effects to the light entities.

### CODE REVIEW

**Comment by hanskroner** [comment]
@Kane610 just in case you hadn't seen the PR. Any chance you could have a look at this?

**Review by Kane610 — CHANGES_REQUESTED** [review]
Sorry, I have been occupied with other things. Please add a new test for this

**Comment by hanskroner** [comment]
@Kane610 - Tests have been updated and the branch rebased on top of the last dependency bumps.

**Review by Kane610 — CHANGES_REQUESTED** [review]
Nearly there

Please add a test similar to test_enum_light_properties where the different capability keys are missing

```
---------- coverage: platform linux, python 3.12.6-final-0 -----------
Name                                           Stmts   Miss  Cover   Missing
----------------------------------------------------------------------------
pydeconz/models/light/light.py                   183      1    99%   330
```

**Comment by hanskroner** [comment]
I hope I understood correctly. A missing `capabilities` key is now covered by the tests:

```
---------- coverage: platform linux, python 3.12.6-final-0 -----------
Name                                                              Stmts   Miss  Cover   Missing
-----------------------------------------------------------------------------------------------
/workspaces/deconz/pydeconz/models/light/light.py                   183      0   100%
```

File: pydeconz/models/light/light.py [hunk 1]
```diff
@@ -215,9 +236,12 @@ class LightEffect(enum.StrEnum):
 
     CANDLE = "candle"
     FIREPLACE = "fireplace"
-    LOOP = "loop"
```
Comment (Kane610) [inline pydeconz/models/light/light.py:218]: Why did you remove loop?

File: tests/lights/test_light.py [hunk 2]
```diff
@@ -298,9 +347,33 @@ async def test_enum_light_properties_no_key(deconz_light):
...
+    assert light.supported_effects is None
     assert light.alert is None
     assert light.color_capabilities is None
     assert light.color_mode is None
     assert light.effect is None
     with pytest.raises(KeyError):
         assert light.fan_speed
+
+
+async def test_enum_light_properties_no_capabilities_color_key(deconz_light):
```
Comment (Kane610) [inline tests/lights/test_light.py:359]: Combine the tests to one using pytest parametrize to modify the expected payload.

**Review by Kane610 — CHANGES_REQUESTED** [review]
Almost there

File: pydeconz/models/light/light.py (hunk 1 above)
Comment (hanskroner) [reply pydeconz/models/light/light.py:218]: `loop` has been renamed `prism` but it shouldn't have been removed - thanks for catching this.

**Comment by hanskroner** [comment]
@Kane610 - I appreciate the guidance and the patience, thank you!

File: pydeconz/models/light/light.py [hunk 3]
```diff
@@ -216,8 +237,12 @@ class LightEffect(enum.StrEnum):
     CANDLE = "candle"
     FIREPLACE = "fireplace"
     LOOP = "loop"
```
Comment (Kane610) [inline pydeconz/models/light/light.py:239]: Could you add a note on loop that it has been replaced with prism? Do you know from what hue fw its been replaced?

```suggestion
    LOOP = "loop"
    """Replaced by prism on later Hue firmwares."""
```

File: tests/lights/test_light.py [hunk 4]
```diff
@@ -298,9 +347,25 @@ async def test_enum_light_properties_no_key(deconz_light):
...
+    ({"capabilities": {"color": {}}}),
+]
+
+
+@pytest.mark.parametrize(("data"), ENUM_CAPABILITIES_DATA)
+async def test_enum_light_properties_no_capabilities(deconz_light, data):
+    """Verify enum properties return expected values or None."""
+    light = await deconz_light({"config": {}, "state": {}, "type": "Color light"})
+
+    light.update(data)
```
Comment (Kane610) [inline tests/lights/test_light.py:370]: No need to test the update method?
```suggestion
    light = await deconz_light({"config": {}, "state": {}, "type": "Color light"} | data)
```

**Review by Kane610 — CHANGES_REQUESTED** [review]
Good job! I think final changes

File: pydeconz/models/light/light.py (hunk 3 above)
Comment (hanskroner) [reply pydeconz/models/light/light.py:239]: I'm not exactly sure on the firmware version, it seems it depends on the kind of light and wasn't rolled out to all lights at the same time. I do have a reference to when deCONZ renamed the effect - I'll add a note referencing [that](https://github.com/dresden-elektronik/deconz-rest-plugin/pull/7206)

File: tests/lights/test_light.py (hunk 4 above)
Comment (hanskroner) [reply tests/lights/test_light.py:370]: I don't think so. It's the same method `test_enum_light_properties` and `test_light_light` are calling to update the state of the light test fixture. I'm not sure how one would go about writing a test for that.

Comment (hanskroner) [reply tests/lights/test_light.py]: That's neat! I didn't think to do that.

File: pydeconz/models/light/light.py [hunk 5]
```diff
@@ -294,6 +322,17 @@ def brightness(self) -> int | None:
...
+    @property
+    def supported_effects(self) -> list[LightEffect] | None:
+        """List of effects supported by a light."""
+        if (
+            "capabilities" in self.raw
+            and "color" in self.raw["capabilities"]
+            and "effects" in self.raw["capabilities"]["color"]
+        ):
+            return list(map(LightEffect, self.raw["capabilities"]["color"]["effects"]))
+        return None
```
Comment (Kane610) [inline pydeconz/models/light/light.py:334]: One final observation, I wonder if this should always return a list

Comment (hanskroner) [reply pydeconz/models/light/light.py:334]: I feel `None` expresses "I don't support effects" better than an empty list ("I don't have any effects").

The HomeAssistant integration also makes the distinction - an empty list of effects shows an empty "Effects" button, while `None` won't show the "Effects" button at all.  The integration could always check if the list is empty and return `None` but, again, I feel the library differentiating between "don't support effects" and "don't have any effects" is more expressive.

Comment (Kane610) [reply pydeconz/models/light/light.py:334]: Ok, lets keep it as is

**Review by Kane610 — APPROVED** [review]

**Comment by Kane610** [comment]
I can do a release tonight of pydeconz

**Comment by Kane610** [comment]
I've release v117
